packages/@aws-cdk/cdk/lib/aspects/README.md

```diff
@@ -0,0 +1,345 @@
+
+```ts
+import cdk = require('@aws-cdk/cdk');
+
+const app = new cdk.App();
+const theBestStack = new cdk.Stack(app, 'MarketingSystem');
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest'));
+theBestStack.apply(new cdk.RemoveTag('StackType', 'TheBest'));
+// now nothing will be tagged
+```
+
+#### Tag Priority
+
+The final control you have is to set a priority number. Tags with highest
+priority number will always win. If tags have the same priority number the tag
+closest in tree distance to the resource will win.
+
+The default priority of `Tag` is 0. The default priority of `RemoveTag` is 1.
+
+```ts
+import cdk = require('@aws-cdk/cdk');
+import ec2 = require('@aws-cdk/aws-ec2');
+
+const app = new cdk.App();
+const theBestStack = new cdk.Stack(app, 'MarketingSystem');
+
+theBestStack.apply(new cdk.Tag('StackType', 'TheBest', {prioirty: 1}));
+
+const vpc = new ec2.VpcNetwork(marketingStack, 'MarketingVpc', {
+  maxAZs: 3, 
+});
+
+// Tag default is 0, so this has no affect
+vpc.apply(new cdk.Tag('StackType', 'TheBestVpc'));
+// RemoveTag default is 1, so this removes the tag from the VPC and children of
+// VPC (e.g. Subnet, RouteTable, NAT Gateway, etc)
+vpc.apply(new cdk.RemoveTag('StackType'));
+```
+
+You can use priority to control tagging regardless of the hierarchy of the tree.
+
+## Aspects
+
+Aspects are a mechanism to extend the CDK without having to directly impact the
+class hierarchy directly. We have implemented Aspects using the [Visitor
+Pattern](https://en.wikipedia.org/wiki/Visitor_pattern).
+
+### Aspects in the CDK
+
+An aspect in the CDK is defined by the interface:
+
+```ts
+/**
+ * Represents an Aspect
+ */
+export interface IAspect {
+  /**
+   * All aspects can visit an IConstruct
+   */
+  visit(node: IConstruct): void;
+}
+```
+
+Aspects can be applied to any `Construct`. During the tree
+preparation phase the aspect will visit each construct in the tree one
+time. Aspects are invoked in the order they were added to the `Construct`,
+starting at the `App` (root of the tree) and progressing in order to the leaf
+nodes (most commonly the CloudFormation Resource). Aspect authors will implement 
+the `visit(IConstruct)` function can inspect the `Construct` for specific characteristics.
+Such as, is this construct a CloudFormation Resource?
+
+### Tag and Remove Tag Aspect
+
+Tag support is the first use of Aspects. The goal was to enable the ability to
+define tags in one place and them have applied consistently for all resources
+that support tagging. In addition the developer should not have to know if the
+resource supports tags. The developer defines the tagging intents for all
+resources within a path. If the resources support tags they are added, else no
+action is taken.
+
+In order to enable additional controls a Tag Aspect can specifically include or
+exclude a CloudFormation Resource Type. See the TagAspectProps interface for
+more details. 
+
+#### Tag Example with ECS
```

Comment:
This should probably go to `examples` and only referenced here

Reply:
Let's keep it simple. We now support tags. You can:

- Add (apply?) a tag, either to specific resources or all but specific resources
- Remove a tag, again either from specific resources or all but specific resources

Tags are inherited, based on the scope. If you tag resource A, and A contains resource B, resource B inherits the tag.

And then code examples, followed by some API-ish text and a link to the reference.